### PR TITLE
Support v2 Linter API

### LINKED ISSUE #7: Update to linter API v2
https://github.com/steelbrain/linter/blob/master/docs/guides/upgrading-to-standard-linter-v2.md

### PR DESCRIPTION
**Pull request by vzamanillo**
Fixes #7

### CODE REVIEW

**Comment by vzamanillo** [comment]
Tests are failing due to an outdatet distro..

**Comment by vzamanillo** [comment]
CircleCi is waiting for status, is this related to branch workflow settings?

**Comment by Arcanemagus** [comment]
Updated the branch protection, looking over the changes now.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,19 +1,108 @@
...
+          - v1-dependencies
+      # Save project state for next steps
+      - persist_to_workspace:
+          root: /tmp
+          paths:
+            - project
+  lint:
+    <<: *defaults
+    docker:
+      - image: circleci/node:latest
```
Comment (Arcanemagus) [inline .circleci/config.yml]: Change this to `node:lts` to be on the slightly safer side.

File: package.json [hunk 2]
```diff
@@ -24,17 +24,21 @@
...
     "linter": {
       "versions": {
-        "1.0.0": "provideLinter"
+        "2.0.0": "provideLinter"
       }
     }
   },
+  "scripts": {
+    "lint": "eslint .",
+    "test": "apm test"
```
Comment (Arcanemagus) [inline package.json]: Although this still works, the preferred method is now `atom --test spec`.

File: spec/linter-gfortran-spec.js [hunk 3]
```diff
@@ -15,17 +15,15 @@ describe('The Fortran provider for AtomLinter', () => {
     waitsForPromise(() => atom.workspace.open(path.join(FILES, 'error.f95')).then(editor =>
       lint(editor).then((messages) => {
         expect(messages.length).toEqual(1);
-        expect(messages[0].type).toEqual('Error');
-        expect(messages[0].text).toEqual('Unterminated character constant beginning at (1)');
-      }),
-    ));
+        expect(messages[0].severity).toEqual('error');
```
Comment (Arcanemagus) [inline spec/linter-gfortran-spec.js]: `toBe`

File: spec/linter-gfortran-spec.js [hunk 4]
```diff
@@ -15,17 +15,15 @@ describe('The Fortran provider for AtomLinter', () => {
...
+        expect(messages[0].excerpt).toEqual('Unterminated character constant beginning at (1)');
```
Comment (Arcanemagus) [inline spec/linter-gfortran-spec.js]: `toBe`

**Pull request merged**